File: moviematch/src/main/java/com/brademobackend/moviematch/model/Usuario.java
package com.brademobackend.moviematch.model;

import java.time.LocalDate;
import java.util.List;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToMany;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id_usuario;
    private String nome_usuario;
    private String email_usuario;
    private String nomeDeUsuario_usuario;
    private String senha_usuario;
    private String fotoPerfil_usuario; //salva o caminho da imagem
    private LocalDate dataNascimento_usuario;

    @ManyToMany
    private List<Filme> filmesCurtidos;
}

File: projeto_brademo/lib/model/Usuario.dart
class Usuario {
  final int idUsuario;
  final String nome;
  final String email;
  final String nomeUsuario;
  final String senha;
  final String fotoPerfil;
  final String dataNascimento;

  Usuario({
    required this.idUsuario,
    required this.nome,
    required this.email,
    required this.nomeUsuario,
    required this.senha,
    required this.fotoPerfil,
    required this.dataNascimento,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      idUsuario: json['id_usuario'],
      nome: json['nome_usuario'],
      email: json['email_usuario'],
      nomeUsuario: json['nomeDeUsuario_usuario'],
      senha: json['senha_usuario'],
      fotoPerfil: json['fotoPerfil_usuario'] ?? '',
      dataNascimento: json['dataNascimento_usuario'],
    );
  }
}